src/omnisharp/extension.ts

```diff
@@ -92,6 +93,13 @@ export async function activate(context: vscode.ExtensionContext, packageJSON: an
         localDisposables.add(forwardChanges(server));
         localDisposables.add(trackVirtualDocuments(server, eventStream));
         localDisposables.add(vscode.languages.registerFoldingRangeProvider(documentSelector, new StructureProvider(server, languageMiddlewareFeature)));
+
+        // Do not set during tests
+        if (process.env.OSVC_SUITE === undefined && vscode.languages.registerDocumentSemanticTokensProvider !== undefined) {
+            const semanticTokensProvider = new SemanticTokensProvider(server, languageMiddlewareFeature);
+            localDisposables.add(vscode.languages.registerDocumentSemanticTokensProvider(documentSelector, semanticTokensProvider, semanticTokensProvider.getLegend()));
```

Comment:
I believe these will explode in a published extension. Aka, I don't believe you're allowed to publish an official extension with proposed APIs unless approved by VSCode folks; we've added TypeScripty equivalent ifdefs to ensure we don't explode in those circumstances.

Reply:
Thanks, I'll have to check on that.